bb8/src/inner.rs

```diff
@@ -158,7 +159,7 @@ where
                 }
                 let approvals = locked.dropped(1, &self.inner.statics);
                 self.spawn_replenishing_approvals(approvals);
-                self.inner.notify.notify_waiters();
+                self.inner.notify.notify_one();
```

Comment:
Nothing broke when I changed it independently. I can move this into its own commit

Reply:
I pulled this into its own PR just to make it clear the tests still pass - https://github.com/djc/bb8/pull/225. It's not necessary for this fix since it still holds even if we bounce all the waiters.